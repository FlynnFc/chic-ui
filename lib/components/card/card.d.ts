import React from 'react';
export interface StyledCardProps {
    width?: string;
    height?: string;
    className?: string;
    src: string;
}
export declare const Image: React.ForwardRefRenderFunction<HTMLDivElement, StyledCardProps>;
declare const _default: React.ForwardRefExoticComponent<StyledCardProps & React.RefAttributes<HTMLDivElement>>;
export default _default;
//# sourceMappingURL=image.d.ts.map