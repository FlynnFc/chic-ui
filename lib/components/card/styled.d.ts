import { StyledCardProps } from '.';
export declare const StyledCard: import("styled-components").StyledComponent<"img", any, StyledCardProps, never>;
//# sourceMappingURL=styled.d.ts.map