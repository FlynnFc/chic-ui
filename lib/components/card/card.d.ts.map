{"version":3,"file":"card.d.ts","sourceRoot":"","sources":["../../../src/components/card/card.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAG1B,MAAM,WAAW,UAAU;IACzB,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,OAAO,CAAC,EAAE,OAAO,CAAC;IAClB,MAAM,CAAC,EAAE,OAAO,CAAC;IACjB,OAAO,CAAC,EAAE,OAAO,CAAC;IAClB,GAAG,EAAE,MAAM,CAAC;CACb;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,wBAAwB,CAAC,gBAAgB,EAAC,UAAU,CAuB7E,CAAC;;AAEF,wBAAqE"}