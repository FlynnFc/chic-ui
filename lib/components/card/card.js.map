{"version":3,"file":"card.js","sourceRoot":"","sources":["../../../src/components/card/card.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,EAAE,WAAW,EAAE,MAAM,UAAU,CAAC;AAYvC,MAAM,CAAC,IAAM,KAAK,GAAgE,UAAC,KAAK,EAAE,GAAG;IAEzF,IAAA,KAOE,KAAK,MAPO,EAAd,KAAK,mBAAG,MAAM,KAAA,EACd,KAME,KAAK,OANQ,EAAf,MAAM,mBAAG,MAAM,KAAA,EACf,SAAS,GAKP,KAAK,UALE,EACT,GAAG,GAID,KAAK,IAJJ,EACH,KAGE,KAAK,QAHQ,EAAf,OAAO,mBAAG,KAAK,KAAA,EACf,KAEE,KAAK,OAFO,EAAd,MAAM,mBAAG,KAAK,KAAA,EACd,KACE,KAAK,QADQ,EAAf,OAAO,mBAAG,KAAK,KAAA,CACP;IAEV,OAAO,CACL,oBAAC,WAAW,IACV,GAAG,EAAE,GAAG,EACR,GAAG,EAAE,GAAG,EACR,KAAK,EAAE,KAAK,EACZ,MAAM,EAAE,MAAM,EACd,SAAS,EAAE,SAAS,EACpB,OAAO,EAAE,OAAO,EAChB,MAAM,EAAE,MAAM,EACd,OAAO,EAAE,OAAO,GAChB,CACH,CAAC;AACJ,CAAC,CAAC;AAEF,eAAe,KAAK,CAAC,UAAU,CAA+B,KAAK,CAAC,CAAC"}