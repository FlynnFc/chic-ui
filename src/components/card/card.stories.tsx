import { Story, Meta } from '@storybook/react';
import Card, { StyledCardProps } from '.';

export default {
  title: 'Components/Card',
  component: Card
} as Meta;

const Template: Story<StyledCardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  href: 'https://www.youtube.com/watch?v=3v4UwjXI4Q0&list=RDMM&index=16',
  hrefText: 'Visit the site',
  height: '200px',
  width: '300px',
  title: 'Title',
  subTitle: 'Sub Title',
  footer: 'Footer',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
};
