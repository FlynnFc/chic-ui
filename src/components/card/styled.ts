import styled from 'styled-components';
import { StyledCardProps } from '.';

export const StyledCard = styled.div<StyledCardProps>`
  display: inline-block;
  box-sizing: border-box;
  color: #0018cf;
  text-decoration: none;
  margin: 0;
  padding: 0;
  font-size: 15px;
  height: ${(pr) => pr.height};
  width: ${(pr) => pr.width};
`;

export const StyledImage = styled.img`
  height: ${(pr) => pr.height};
  width: ${(pr) => pr.width};
  object-fit: cover;
`;

