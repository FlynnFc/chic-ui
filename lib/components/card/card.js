import React from 'react';
import { StyledImage } from './styled';
export var Card = function (props, ref) {
    var _a = props.width, width = _a === void 0 ? '100%' : _a, _b = props.height, height = _b === void 0 ? '100%' : _b, className = props.className, src = props.src;
    return (React.createElement(StyledImage, { src: src, ref: ref, width: width, height: height, className: className }));
};
export default React.forwardRef(Card);
//# sourceMappingURL=image.js.map