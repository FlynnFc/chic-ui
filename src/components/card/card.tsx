import React, { ReactNode } from 'react';
import { themeType } from 'src/config/themes';
import { StyledCard, StyledDescription, StyledFooter, StyledFooterText, StyledFooterLink, StyledTitle, StyledSubTitle, StyledImage } from './styled';

export interface StyledCardProps {
  align?:string,
  children?: ReactNode;
  href?: string;
  width?: string;
  height?: string;
  className?: string;
  src?: string;
  title?: string;
  subTitle?: ReactNode;
  footer?: ReactNode;
  description?: string;
  hrefText?: string;
  type?: themeType;
}

export const Card: React.ForwardRefRenderFunction<
  HTMLDivElement,
  StyledCardProps
> = (props, ref) => {
  const { 
    className, 
    href, 
    src, 
    width = '300px',
    height = '200px',
    title,
    subTitle,
    children,
    footer,
    hrefText,
    description,
    type= 'light'
    } = props;

  return (
    <StyledCard
      className={className}
      ref={ref}
      width={width}
      height={height}
      type={type}
    >
      {src && (
        <StyledImage src={src} width={'100%'}
      height={'100%'}/>
      ) }      
      {title && <StyledTitle children={title}/>}
      {subTitle && <StyledSubTitle children={subTitle}/>}
      {description && <StyledDescription children={description}/>}
      {footer && <StyledFooter>
        <StyledFooterText children={footer}/>
        {href && <StyledFooterLink href={href} children={hrefText}/>}
      </StyledFooter> }
      {children}
    </StyledCard>
  );
};

export default React.forwardRef<HTMLDivElement, StyledCardProps>(Card);
