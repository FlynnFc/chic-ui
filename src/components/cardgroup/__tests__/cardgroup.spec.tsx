import CardGroup from '..';
import { Card } from '../../card';
import { render } from '../../../test/utils';

describe('Card Group Component', () => {
  it('Renders Correctly', () => {
    const CardGroups = render(
      <>
      <CardGroup>
      <Card
          type="secondary"
          title="Title"
          subTitle="Subtitle"
          description="Lorem ipsum dolor sit amet, consectetur"
          footer="Footer"
          href="https://github.com/karishmashuklaa/chic-ui"
          hrefText="Visit site"
        />
      </CardGroup>
      </>
    );
    expect(CardGroups).toMatchSnapshot();
  });
});
