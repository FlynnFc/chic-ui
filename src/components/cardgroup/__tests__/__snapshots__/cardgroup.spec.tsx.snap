// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Group Component Renders Correctly 1`] = `
<div
  className="sc-bdfBQB bnRANJ"
>
  <div
    className="sc-jrAFXE gJbdtU"
  >
    <div
      className="sc-kEjbQP ehWzNF"
      type="secondary"
    >
      <h2
        className="sc-crrszt gpdIdo"
      >
        Title
      </h2>
      <h5
        className="sc-dQoVA dvXURs"
      >
        Subtitle
      </h5>
      <p
        className="sc-bqyKOL jhGSPc"
      >
        Lorem ipsum dolor sit amet, consectetur
      </p>
      <div
        className="sc-kstqJO lhaulb"
      >
        <p
          className="sc-hBEYId fPQBgu"
        >
          Footer
        </p>
        <a
          className="sc-fodVek bxlGRR"
          href="https://github.com/karishmashuklaa/chic-ui"
        >
          Visit site
        </a>
      </div>
    </div>
  </div>
</div>
`;
